packages/support/src/Concerns/Macroable.php

```diff
@@ -39,6 +39,11 @@ public static function flushMacros(): void
         static::$macros = [];
     }
 
+    public static function hasMacro(string $name): bool
+    {
+        return isset(static::$macros[$name]);
```

Comment:
This will return true even if the macro is registered for a different class

Reply:
Apologies, you are right indeed. I just copied the implementation from Laravel assuming Filament had the same. I updated the test to now account for this and also check macros on child components.

Reply:
The reason we have our own trait instead of using Laravel's is because it handles inherited classes much better